folly/io/async/AsyncSocket.cpp

```diff
@@ -662,7 +662,11 @@ void AsyncSocket::writeChain(WriteCallback* callback, unique_ptr<IOBuf>&& buf,
   constexpr size_t kSmallSizeMax = 64;
   size_t count = buf->countChainElements();
   if (count <= kSmallSizeMax) {
+    // suppress "warning: variable length array ‘vec’ is used [-Wvla]"
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wvla"
```

Comment:
We have that warning enabled enabled even internally. Completely unsetting it will cause it to no longer compile. It has to be set to either 0 or 1.